src/nwb/NWBFile.hpp

```diff
@@ -46,26 +46,33 @@ class NWBFile
    * @brief Initializes the NWB file by opening and setting up the file
    * structure.
    */
-  void initialize();
+  Status initialize();
 
   /**
    * @brief Finalizes the NWB file by closing it.
    */
-  void finalize();
+  Status finalize();
 
   /**
    * @brief Create ElectricalSeries objects to record data into.
    * Created objects are stored in recordingContainers.
+   * Note, this function will fail if the file is in a mode where
+   * new objects cannot be added.
```

Comment:
```suggestion
   * Note, this function will fail if the file is in a mode where
   * new objects cannot be added which can be checked via
   * nwbfile.io->canModifyObjects()
```

Reply:
I would also add a similar comment to the other `create` methods where this applies